### PR TITLE
Nullable annotate System.Security.Cryptography.OpenSSL

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Contributes to https://github.com/dotnet/runtime/issues/2339

### CODE REVIEW

**Review by bartonjs — APPROVED** [review]

File: src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ internal static partial class DSAImplementation
         public sealed partial class DSAOpenSsl : DSA
         {
             private const int BitsPerByte = 8;
-            private Lazy<SafeDsaHandle> _key;
+            private Lazy<SafeDsaHandle> _key = null!;
```
Comment (stephentoub) [inline src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs]: Why is this necessary?  Are there ctors that don't set this in other files?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs (hunk 1 above)
Comment (eiriktsarpalis) [reply src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs]: You're right, it isn't. Not sure I remember why I did this.

Comment (eiriktsarpalis) [reply src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs]: Actually turns out that it's required for the OSX build. Reverting...

Comment (stephentoub) [reply src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs]: And it can't be handled in the macOS-specific files?

Comment (eiriktsarpalis) [reply src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs]: The only way I can think of is adding an extra `_key = null!;` assignment in  the offending constructors.

Comment (stephentoub) [reply src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs]: Does that build not use this field at all? I can look at the code later, but is making this non-nullable actually wrong for objects built with those ctors?

Comment (eiriktsarpalis) [reply src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs]: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Security.Cryptography.OpenSsl/src/System/Security/Cryptography/DSAOpenSsl.cs#L43 

It's the usual problem, these additional ctors are initializing the field using a helper method.

Comment (eiriktsarpalis) [reply src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs]: Note that both files are named `DSAOpenSSl.cs`.

Comment (stephentoub) [reply src/libraries/Common/src/System/Security/Cryptography/DSAOpenSsl.cs]: > It's the usual problem, these additional ctors are initializing the field using a helper method

Ok
